monitor.cpp

```diff
@@ -3,36 +3,54 @@
 #include <thread>
 #include <chrono>
 #include <iostream>
+#include <functional>
+#include <vector>
+
 using std::cout, std::flush, std::this_thread::sleep_for, std::chrono::seconds;
+using std::function;
+using std::string;
+using std::vector;
 
-int vitalsOk(float temperature, float pulseRate, float spo2) {
-  if (temperature > 102 || temperature < 95) {
-    cout << "Temperature is critical!\n";
-    for (int i = 0; i < 6; i++) {
-      cout << "\r* " << flush;
-      sleep_for(seconds(1));
-      cout << "\r *" << flush;
-      sleep_for(seconds(1));
-    }
-    return 0;
-  } else if (pulseRate < 60 || pulseRate > 100) {
-    cout << "Pulse Rate is out of range!\n";
+void displayCriticalAlert(const std::string& message) {
+    cout << message << "\n";
     for (int i = 0; i < 6; i++) {
-      cout << "\r* " << flush;
-      sleep_for(seconds(1));
-      cout << "\r *" << flush;
-      sleep_for(seconds(1));
+        cout << "\r* " << flush;
+        sleep_for(seconds(1));
+        cout << "\r *" << flush;
+        sleep_for(seconds(1));
     }
-    return 0;
-  } else if (spo2 < 90) {
-    cout << "Oxygen Saturation out of range!\n";
-    for (int i = 0; i < 6; i++) {
-      cout << "\r* " << flush;
-      sleep_for(seconds(1));
-      cout << "\r *" << flush;
-      sleep_for(seconds(1));
+}
+
+bool isTemperatureCritical(float temperature) {
+    return temperature > 102 || temperature < 95;
+}
+
+bool isPulseRateOutOfRange(float pulseRate) {
+    return pulseRate < 60 || pulseRate > 100;
+}
+
+bool isSpo2OutOfRange(float spo2) {
+    return spo2 < 90;
+}
+
+int vitalsOk(float temperature, float pulseRate, float spo2) {
+    // Define a vector of functions that perform the vital checks
+    vector<std::tuple<function<bool()>, string>> checks = {
```

Comment:
Nice approach to keep a vector of processors :)
One suggestion for simplification: The value of temperature, pulse and spo2 are already known at this point. So we can keep a vector of `tuple<bool, string>`, and initialize the booleans by calling the function directly.
Your approach would be needed for lazy evaluation - like if we didn't know the values at that point in the code, and would pass it to the function later.

Reply:
Yes I understood that here we need all the values so if i use the above approach it will return as soon as it will find temperature but that is not the case.